### PR TITLE
Update destructuring 2.md

### PR DESCRIPTION
**Pull request by SRGOM**
I'm very new to Rust and the changes I proposed may either be incorrect or not fit in the flow of the document. Please feel free to suggest modifications that I can make or to reject altogether. Thanks.

### CODE REVIEW

File: destructuring 2.md [hunk 1]
```diff
@@ -45,14 +45,10 @@ semantics:
...
-object in Rust has a destructor if it implements the `Drop` trait. Just like
-C++, the destructor is executed just before an object is destroyed. If an object
-has a destructor then it has move semantics. If it does not, then all of its
-fields are examined and if any of those do then the whole object has move
-semantics. And so on down the object structure. If no destructors are found
-anywhere in an object, then it has copy semantics.
+You can also choose to have copy semantics for structs defined by you. 
+Use `#[derive(Copy)]` to implement the `Copy` trait and the struct would 
+have Copy semantics. You may also need to derive the `Clone` trait needed
+by the `Copy` trait.
```
Comment (nrc) [inline destructuring 2.md]: This is an improvement but it is not quite complete. So, I would say that to have copy semantics you have to implement `Copy` and `derive` is one way to do that. It is also worth noting that there restrictions on whether a type can have the `Copy` trait implemented - in particular, the destructor thing (see https://doc.rust-lang.org/std/marker/trait.Copy.html#when-can-my-type-be-copy for details).

**Comment by SRGOM** [comment]
Thanks you for taking the time to comment (and of course the very wonderfully written series). I (think I) have addressed your comment and put back some destructor related content from my previous deletion. Please let me know if any more changes are required.

File: destructuring 2.md [hunk 2]
```diff
@@ -45,14 +45,14 @@ semantics:
...
 
-Rust determines if an object has move or copy semantics by looking for
-destructors. Destructors probably need a post of their own, but for now, an
-object in Rust has a destructor if it implements the `Drop` trait. Just like
-C++, the destructor is executed just before an object is destroyed. If an object
-has a destructor then it has move semantics. If it does not, then all of its
-fields are examined and if any of those do then the whole object has move
-semantics. And so on down the object structure. If no destructors are found
-anywhere in an object, then it has copy semantics.
+You can also choose to have copy semantics for user-defined structs
```
Comment (nrc) [inline destructuring 2.md]: s/structs/types

File: destructuring 2.md [hunk 3]
```diff
@@ -45,14 +45,14 @@ semantics:
...
+by implementing the `Copy` trait. One straightforward way to do that is 
+to add `#[derive(Copy)]` before the definition of the struct. Not all
```
Comment (nrc) [inline destructuring 2.md]: and here

Comment (nrc) [inline destructuring 2.md]: and below in a few places

File: destructuring 2.md [hunk 4]
```diff
@@ -45,14 +45,14 @@ semantics:
...
+user-defined structs are allowed to implement `Copy` trait, in particular 
+those  with a destructor. Destructors probably need a post of their own,
+but for now, an object in Rust has a destructor if it implements the `Drop`
+trait. Just like C++, the destructor is executed just before an object is 
+destroyed. If an object has a destructor then it has move semantics.
```
Comment (nrc) [inline destructuring 2.md]: I would drop this last sentence

**Comment by SRGOM** [comment]
I've addressed your comments, please let me know if any other changes are required.

**Comment by nrc** [comment]
Thanks for the changes, looks good!

**Pull request merged**